ui/cockpit/src/legacy/client/scripts/components/variables/variable-add-dialog.js

```diff
@@ -97,35 +100,56 @@ var Controller = [
 
       var data = angular.extend({}, newVariable);
 
-      if (data.type === "Date") {
-        data.value = fixDate(data.value);
-      }
+      if (operation.variables) {
+        operation.variables = operation.variables.filter(variable => {
+          return data.name !== variable.variable.name;
+        });
 
-      var instanceAPI = isProcessInstance ? processInstance : caseInstance;
-      instanceAPI
-        .setVariable(instance.id, data)
-        .then(function() {
-          $scope.status = SUCCESS;
-
-          Notifications.addMessage({
-            status: $translate.instant("VARIABLE_ADD_MESSAGE_STATUS_FINISHED"),
-            message: $translate.instant("VARIABLE_ADD_MESSAGE_MESSAGE_ADD", {
-              name: data.name
-            }),
-            exclusive: true
-          });
-        })
-        .catch(function(data) {
-          $scope.status = FAIL;
-
-          Notifications.addError({
-            status: $translate.instant("VARIABLE_ADD_MESSAGE_STATUS_FINISHED"),
-            message: $translate.instant("VARIABLE_ADD_MESSAGE_MESSAGE_ERROR", {
-              message: data.message
-            }),
-            exclusive: true
-          });
+        operation.variables.push({
+          variable: data,
+          valid: true,
+          changed: false
```

Comment:
This is okay for the current architecture. When building this in React, we should avoid manipulating data directly from the display component and work with callbacks.